### DISCUSSION TITLE
when is use vscode-languageclient in nextjs, builder warning Critical dependency

### DISCUSSION THREAD

**Discussion by todwang** [vercel/next.js#48614] [Help]
### Summary

here is the log:
`
../../node_modules/.pnpm/vscode-languageserver-types@3.17.3/node_modules/vscode-languageserver-types/lib/umd/main.js
Critical dependency: require function is used in a way in which dependencies cannot be statically extracted

Import trace for requested module:
../../node_modules/.pnpm/vscode-languageserver-types@3.17.3/node_modules/vscode-languageserver-types/lib/umd/main.js
../../node_modules/.pnpm/vscode-languageserver-protocol@3.17.3/node_modules/vscode-languageserver-protocol/lib/common/api.js
../../node_modules/.pnpm/vscode-languageserver-protocol@3.17.3/node_modules/vscode-languageserver-protocol/lib/node/main.js
../../node_modules/.pnpm/vscode-languageclient@8.1.0/node_modules/vscode-languageclient/lib/common/api.js
../../node_modules/.pnpm/vscode-languageclient@8.1.0/node_modules/vscode-languageclient/lib/browser/main.js
../../node_modules/.pnpm/vscode-languageclient@8.1.0/node_modules/vscode-languageclient/browser.js
`

next,js does not seem to like the UMD module it found
but how to fixed this warning?

### Additional information

```js
"next": "13.3.0",
    "react": "18.2.0",
    "react-dom": "18.2.0",
```

### Example

https://github.com/todwang/my-app.git

**Comment by AKclown** [comment]
Any progress, I'm having the same problem. When I delete the .lock and reinstall, this problem occurs
![image](https://github.com/vercel/next.js/assets/33648653/bb4d141a-2fc2-4e59-b480-4762852f4989)
